### PR TITLE
fixed issue 116 for pagination

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

**Comment by m-kovalsky** [comment]
#116

File: src/sempy_labs/lakehouse/_get_lakehouse_tables.py [hunk 1]
```diff
@@ -108,7 +108,9 @@ def get_lakehouse_tables(
                 "Location": i.get("location"),
             }
             dfs.append(pd.DataFrame(new_data, index=[0]))
-    df = pd.concat(dfs, ignore_index=True)
+
+    if dfs:
```
Comment (eisber) [inline src/sempy_labs/lakehouse/_get_lakehouse_tables.py:112]: should you maybe return straight from here if you don't have rows?
are the extended/count_rows/... paths going to do what you want?

**Review by eisber — CHANGES_REQUESTED** [review]

File: src/sempy_labs/lakehouse/_get_lakehouse_tables.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/lakehouse/_get_lakehouse_tables.py:112]: updated accordingly

File: src/sempy_labs/lakehouse/_get_lakehouse_tables.py [hunk 2]
```diff
@@ -96,6 +95,9 @@ def get_lakehouse_tables(
 
     responses = pagination(client, response)
 
+    if not responses[0].get("data"):
```
Comment (eisber) [inline src/sempy_labs/lakehouse/_get_lakehouse_tables.py:98]: possible that response itself has no elements?

**Review by eisber — CHANGES_REQUESTED** [review]

**Review by eisber — APPROVED** [review]

**Pull request merged**